ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -123,6 +124,9 @@ public class ReplicaThread implements Runnable {
   private final Predicate<MessageInfo> skipPredicate;
   private volatile boolean allDisabled = false;
   private final ReplicationManager.LeaderBasedReplicationAdmin leaderBasedReplicationAdmin;
+  private boolean enableAcyclicReplication;
+  private int maxIterationsPerGroupPerCycle;
+  private volatile boolean shouldTerminateCurrentCycle = false;
```

Comment:
Can you explain the use of this flag and in what scenarios, the value will be false.

Reply:
If this is set true and if current cycle is running, we will not create any new iterations of any group and will wait for existing groups to finish.

Reply:
@snalli check this flag for replica addition and deletion.